test/NuGet.Core.Tests/NuGet.Protocol.Tests/Resources/RepositorySignatureResourceTests.cs

```diff
@@ -0,0 +1,111 @@
+
+            // Act
+            var repositorySignatureResource = await repo.GetResourceAsync<RepositorySignatureResource>();
+
+            // Assert
+            Assert.Null(repositorySignatureResource);
+        }
+
+        [Fact]
+        public void RepositorySignatureResource_NoAllRepositorySigned()
+        {
+            // Arrange
+            var source = "http://unit.test1/v3-with-flat-container/index.json";
+            var responses = new Dictionary<string, string>();
+            var repo = StaticHttpHandler.CreateSource(source, Repository.Provider.GetCoreV3(), responses);
+            var jobject = JObject.Parse(JsonData.repoSignDataNoAllRepositorySigned);
+
+
+            // Act & Assert
+            Assert.Throws<FatalProtocolException>(() => new RepositorySignatureResource(jobject, repo));
+        }
+
+        [Fact]
+        public void RepositorySignatureResource_NoCertInfo()
+        {
+            // Arrange
+            var source = "http://unit.test1/v3-with-flat-container/index.json";
+            var responses = new Dictionary<string, string>();
+            var repo = StaticHttpHandler.CreateSource(source, Repository.Provider.GetCoreV3(), responses);
+            var jobject = JObject.Parse(JsonData.repoSignDataNoCertInfo);
+
+            // Act & Assert
+            Assert.Throws<FatalProtocolException>(() => new RepositorySignatureResource(jobject, repo));
+        }
+
+        public static RepositorySignatureResource GetRepositorySignatureResource()
+        {
+            var certInfo = new Mock<IRepositoryCertInfo>();
+
+            var fingerPrints = new Dictionary<string, string>() { { "2.16.840.1.101.3.4.2.1", "3f9001ea83c560d712c24cf213c3d312cb3bff51ee89435d3430bd06b5d0eece" } };
+
+            certInfo.SetupGet(p => p.Issuer).Returns("CN=DigiCert SHA2 Assured ID Code Signing CA, OU=www.digicert.com, O=DigiCert Inc, C=US");
+            certInfo.SetupGet(p => p.Fingerprints).Returns(new FingerprintInfos(fingerPrints));
+            certInfo.SetupGet(p => p.NotBefore).Returns(DateTime.Parse("2018-02-26T00:00:00.0000000Z"));
+            certInfo.SetupGet(p => p.NotAfter).Returns(DateTime.Parse("2021-01-27T12:00:00.0000000Z"));
+            certInfo.SetupGet(p => p.Subject).Returns("CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US");
+            certInfo.SetupGet(p => p.ContentUrl).Returns("https://api.nuget.org/v3-index/repository-signatures/certificates/3f9001ea83c560d712c24cf213c3d312cb3bff51ee89435d3430bd06b5d0eece.crt");
+
+            var certInfos = new List<IRepositoryCertInfo>() { certInfo.Object };
+            return new RepositorySignatureResource(false, certInfos);
+        }
+
+        public static void VerifyCertInfo(IRepositoryCertInfo certInfo)
+        {
+            Assert.Equal("3f9001ea83c560d712c24cf213c3d312cb3bff51ee89435d3430bd06b5d0eece", certInfo.Fingerprints["2.16.840.1.101.3.4.2.1"]);
```

Comment:
You can dedupe these string literals and make them private constants in this test class.